### PR TITLE
[Slates] Implement DSJSON parser for slates

### PR DESCRIPTION
**Pull request by jackgerrits**

### CODE REVIEW

File: test/unit_test/dsjson_parser_test.cc [hunk 1]
```diff
@@ -19,8 +19,9 @@ multi_ex parse_dsjson(vw& all, std::string line)
       (VW::example_factory_t)&VW::get_unused_example, (void*)&all, &interaction);
 
   multi_ex result;
-  for (size_t i = 0; i < examples.size(); ++i) {
-	  result.push_back(examples[i]);
+  for (size_t i = 0; i < examples.size(); ++i)
```
Comment (lalo) [inline test/unit_test/dsjson_parser_test.cc:22]: nit: for each loop?

**Comment by lokitoth** [comment]
Adding a ping that I am still working on this.

**Comment by lokitoth** [comment]
I think the build break is due to your moving namespaces for slates label?

**Comment by jackgerrits** [comment]
Oh, I didn't notice the break. Thanks!

**Pull request merged**